src/buffer/out/search.cpp

```diff
@@ -17,11 +17,11 @@ using namespace Microsoft::Console::Types;
 // - Once you've found something, you can perform actions like .Select() or .Color()
 // Arguments:
 // - textBuffer - The screen text buffer to search through (the "haystack")
-// - uiaData - The IUiaData type reference, it is for providing selection methods
+// - uiaData - The IRenderData type reference, it is for providing selection methods
 // - str - The search term you want to find (the "needle")
 // - direction - The direction to search (upward or downward)
 // - sensitivity - Whether or not you care about case
-Search::Search(IUiaData& uiaData,
+Search::Search(Microsoft::Console::Render::IRenderData& uiaData,

```

Comment:
The reason they were separate is that... well, Search shouldn't need to know what a "render" is.

Arguably, it should also not need to know what a "UIA" is.

Reply:
Right... It shouldn't have to know about either. Since this is just an internal implementation detail depending on larger interfaces is IMO not at the same as a public API where such concerns might be more important.
But more importantly my goal is merging the conhost and Windows Terminal implementations for `IRenderData` into one. There's also `SCREEN_INFORMATION`, etc. which are similarly quite redundant in our larger code base. I personally want our _entire_ screen-related application state to be serializable so that it is possible to store it on disk (session restore) and snapshottable (not holding locks during rendering). Merging these interfaces is just the first, but required step IMO.